### PR TITLE
Support extended property patterns in 'use recursive' refactoring

### PR DESCRIPTION
**Pull request by alrz**
.. and minor cleanup.

### CODE REVIEW

File: src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs [hunk 1]
```diff
@@ -295,7 +280,7 @@ static SyntaxToken Flip(SyntaxToken token)
                 .Where(d => d.Identifier.ValueText == identifierName.Identifier.ValueText)
                 .FirstOrDefault();
 
-            if (designation is not { Parent: PatternSyntax containingPattern })
+            if (designation is not { Parent: PatternSyntax(not SyntaxKind.ListPattern) containingPattern })
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs:283]: =-o

i did not realize you could do that.

--

also, can you doc why this clause was added?

File: src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs [hunk 2]
```diff
@@ -360,10 +345,21 @@ LessThanOrEqualExpression or
         private static SubpatternSyntax CreateSubpattern(ImmutableArray<IdentifierNameSyntax> names, PatternSyntax pattern)
         {
             Debug.Assert(!names.IsDefaultOrEmpty);
-            var subpattern = Subpattern(names[0], pattern);
-            for (var i = 1; i < names.Length; i++)
-                subpattern = Subpattern(names[i], RecursivePattern(subpattern));
-            return subpattern;
+
+            if (names.Length > 1 && ((CSharpParseOptions)names[0].SyntaxTree.Options).LanguageVersion >= LanguageVersion.CSharp10)
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs:349]: kinda ugly.  i wish thtere was a clean way for you to get this info.

Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs:349]:
```suggestion
            if (names.Length > 1 && names[0].SyntaxTree.Options.LanguageVersion() >= LanguageVersion.CSharp10)
```

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs (hunk 1 above)
Comment (alrz) [reply src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs:284]: Done. 

(this is using Deconstruct(SyntaxNode, out SyntaxKind))

Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs:284]: yup.  i just didn't realize that you could do the `not ...` bit in the position sectino :)  just never occurred to me :)

**Comment by CyrusNajmabadi** [comment]
thanks!

**Pull request merged**